filters/auth/tokeninfo.go

```diff
@@ -353,6 +358,12 @@ func (f *tokeninfoFilter) Request(ctx filters.FilterContext) {
 	}
 
 	authorized(ctx, uid)
+
+	realm, _ := authMap[realmKey].(string)
+	if f.maskRealms.MatchString(realm) {
+		ctx.StateBag()[logfilter.MaskedAuthUserKey] = realm
```

Comment:
This looks wrong according to 
`// MaskedAuthUserKey is the same as AuthUserKey, but some realms are masked`

What do you think about pivoting design in the following way:
1. Use allowlist instead of masklist - i.e. user explicitly specifies for which realms auth user should be set on the span
2. Is there a real necessity to use regexp for realm match? How may realms are there in total afterall (or is it a way to specify a list `(employee|whatever)` maybe)?
3. Get rid of conditional logic here but rather store `realm` along with `auth-user` in statebag (in `auth.go` I guess) and move all the decisions to the filter(s).

Reply:
> yes, I'm still wondering how to do it - I didn't find a realm in https://tools.ietf.org/html/rfc7662#section-2.2. Can you help?

`auth.go` used by all filters to set `auth-user` at the moment is only aware about `username`. One way out may be to extend it and make it also aware of a new concept, say `auth-realm`. Filters that do not support realm may set it to empty or to some predefined constant placeholder.

Reply:
> 1. Use allowlist instead of masklist - i.e. user explicitly specifies for which realms auth user should be set on the span. If not set, realm is not considered.
> 2. Is there a real necessity to use regexp for realm match? How may realms are there in total afterall (or is it a way to specify a list `(employee|whatever)` maybe)?

no strong reason from my side.
I chose this approach after a discussion with @aryszka that masking values might be a concept that can be re-used in other places.

> 3. Get rid of conditional logic here but rather store `realm` along with `auth-user` in statebag (in `auth.go` I guess) and move all the decisions to the `statebagtotag` filter. This would allows building similar logic based on realm in other filters as well.

sounds good